tests/e2e/vsan_stretched_cluster.go

```diff
@@ -0,0 +1,364 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fdep "k8s.io/kubernetes/test/e2e/framework/deployment"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+	fss "k8s.io/kubernetes/test/e2e/framework/statefulset"
+)
+
+var _ = ginkgo.Describe("[vsan-stretch-vanilla] vsan stretched cluster tests", func() {
+	f := framework.NewDefaultFramework("vsan-stretch")
+	var (
+		client            clientset.Interface
+		namespace         string
+		nodeList          *v1.NodeList
+		storagePolicyName string
+		scParameters      map[string]string
+		storageClassName  string
+	)
+
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		bootstrap()
+		var err error
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		readVcEsxIpsViaTestbedInfoJson(GetAndExpectStringEnvVar(envTestbedInfoJsonPath))
+
+		initialiseFdsVar(ctx)
+		err = waitForAllNodes2BeReady(ctx, client, pollTimeout*4)
+		framework.ExpectNoError(err, "cluster not completely healthy")
+
```

Comment:
Can we also check if CSI pods are running fine before starting each test

Reply:
I will add a TODO for now, will address it later